src/org/mirah/jvm/mirrors/generics/xx_type_invoker.mirah

```diff
@@ -100,16 +100,20 @@ class TypeInvoker < BaseSignatureReader
   end
 
   def self.invoke(context:Context, type:MirrorType, args:List,
-                  outerTypeArgs:Map):MirrorType
+                  outerTypeArgs:Map, processed_signatures:Map):MirrorType
     dtype = DeclaredMirrorType(type)
     if dtype.signature.nil? || args.any? {|a| a.nil?}
       type
     else
-      invoker = TypeInvoker.new(context, outerTypeArgs, args)
-      invoker.read(dtype.signature)
+      invoker = TypeInvoker processed_signatures[dtype.signature]
```

Comment:
That's fun. Is this a cast with the optional parens removed? I hadn't considered that you could do that before. :)